examples/v1beta1/pipelineruns/using-optional-workspaces-in-when-expressions.yaml

```diff
@@ -0,0 +1,66 @@
+# This example demonstrates using the workspaces.<name>.bound variable
+# in a when expression to selectively run different portions of a Pipeline
+# based on the presence of an optional workspace.
+#
+# In the PipelineRun below an optional message-of-the-day workspace is accepted
+# by the Pipeline. If that workspace is provided then the print-motd task is
+# executed. If that workspace is not provided then a print-default-motd task
+# is run instead. We supply a ConfigMap for the workspace and so the print-motd
+# task ends up running and printing the contents of each entry in the ConfigMap.
+apiVersion: v1
+kind: ConfigMap
```

Comment:
Not sure how `configmap` is handled in our CI. 

`kubectl apply` followed by `kubectl delete` can not be used with `generateName`. With `kubectl create` and `generateName`, pipelineRun is assigned random name which can be deleted using `kubectl delete pr`  or `tkn pr delete` but leaves the `configmap` as is. Does our cleanup cron job, deletes such config maps? Does this concern even apply in our CI?

Reply:
hm, it looks like we deploy them using `ko apply` all into the same namespace and cluster.

`ko apply` in library.sh: https://github.com/tektoncd/plumbing/blob/ef09ae0cb20c1b320b5e3f7a8707030a50d8be17/scripts/library.sh#L472-L480

i believe we run yaml suite in a brand new cluster though.  so the configmap is created and then quickly deleted when the cluster is cleaned up.